File: java/examples/StreamArticles.java
import io.rtpr.sdk.RtprWebSocket;
import io.rtpr.sdk.model.Article;

import java.util.Arrays;

/**
 * Stream real-time press releases via WebSocket.
 */
public class StreamArticles {
    public static void main(String[] args) {
        RtprWebSocket ws = new RtprWebSocket.Builder()
                .apiKey("your_api_key")
                .listener(new RtprWebSocket.Listener() {
                    @Override
                    public void onConnect() {
                        System.out.println("Connected to RTPR real-time feed");
                    }

                    @Override
                    public void onArticle(Article article) {
                        System.out.printf("[%s] %s%n", article.getTicker(), article.getTitle());
                        System.out.printf("  Source: %s%n", article.getAuthor());
                        System.out.printf("  Body:   %s...%n%n",
                                article.getArticleBody().substring(0, Math.min(120, article.getArticleBody().length())));
                    }

                    @Override
                    public void onDisconnect(int code, String reason) {
                        System.out.printf("Disconnected (code=%d): %s%n", code, reason);
                    }

                    @Override
                    public void onError(Throwable error) {
                        System.err.println("Error: " + error.getMessage());
                    }
                })
                .build();

        // Subscribe to specific tickers, or use Arrays.asList("*") for firehose
        ws.connect(Arrays.asList("AAPL", "TSLA", "NVDA", "MSFT"));

        // Keep the program running
        try {
            Thread.sleep(Long.MAX_VALUE);
        } catch (InterruptedException e) {
            ws.disconnect();
        }
    }
}

File: cpp/examples/stream_articles.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include "rtpr/websocket.hpp"

int main() {
    rtpr::WebSocketClient ws("your_api_key");

    ws.on_connect([]() {
        std::cout << "Connected to RTPR real-time feed\n";
    });

    ws.on_article([](const rtpr::Article& article) {
        std::cout << "[" << article.ticker << "] " << article.title << "\n";
        std::cout << "  Source: " << article.author << "\n";
        auto body_preview = article.article_body.substr(
            0, std::min<size_t>(120, article.article_body.size()));
        std::cout << "  Body:   " << body_preview << "...\n\n";
    });

    ws.on_error([](const std::string& error) {
        std::cerr << "Error: " << error << "\n";
    });

    ws.on_disconnect([](int code, const std::string& reason) {
        std::cout << "Disconnected (code=" << code << "): " << reason << "\n";
    });

    // Subscribe to specific tickers, or {"*"} for firehose
    ws.connect({"AAPL", "TSLA", "NVDA", "MSFT"});

    return 0;
}
